src/txn/txn_rollback_to_stable.c

```diff
@@ -176,6 +176,17 @@ __rollback_abort_update(WT_SESSION_IMPL *session, WT_ITEM *key, WT_UPDATE *first
                 F_CLR(stable_upd, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
             if (tombstone != NULL)
                 F_CLR(tombstone, WT_UPDATE_HS | WT_UPDATE_TO_DELETE_FROM_HS);
+        } else if (WT_IS_HS(session->dhandle) && stable_upd->type != WT_UPDATE_TOMBSTONE) {
```

Comment:
My main concern here is the overhead this pass will add to RTS. Based on the history store size, now switchover from secondary to primary will take proportionally more time.